source/android/Additions/TraceWriter.cs

```diff
@@ -92,159 +89,13 @@ public static void WriteTrace(object exception)
-                                    foreach (Match m in _StackTraceLine.Matches(trace))
-                                    {
-                                        var method = m.Groups[1].Value;
-                                        if (AppNamespaces != null)
-                                        {
-                                            //use an application provided list of classes to determine which namespaces
-                                            //should be mapped into the package name so hockey app can pick out the correct
-                                            //top stacktrace line.  It unfortunately means you end up with an extra copy of
-                                            //package name at the beginning of your c# crash traces
-                                            foreach (var prefix in AppNamespaces)
-                                            {
-                                                if (method.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
-                                                {
-                                                    method = _AppPackage + "." + method;
-                                                    break;
-                                                }
-                                            }
-                                        }
-                                        else
-                                        {
-                                            //default to things that don't match mono/android/java, again, this forces an extra copy of
-                                            //package name at the beginning of your c# crash traces
-                                            if (!method.StartsWith("mono", StringComparison.OrdinalIgnoreCase) &&
-                                                !method.StartsWith("android", StringComparison.OrdinalIgnoreCase) &&
-                                                !method.StartsWith("system", StringComparison.OrdinalIgnoreCase) &&
-                                                !method.StartsWith("java", StringComparison.OrdinalIgnoreCase))
-                                            {
-                                                method = _AppPackage + "." + method;
-                                            }
-                                        }
-                                        //this forces the arguments part to look more like the file line number part that
-                                        //android stack traces normally have
-                                        var arguments = m.Groups[2].Value.Trim();
-                                        arguments = arguments.Replace(' ', '_');
-                                        arguments = arguments.Replace(',', '_');
+                ExceptionHandler.SaveException(Java.Lang.Throwable.FromException(exception as Exception), null, Java.Lang.Thread.CurrentThread(), _Listener);
```

Comment:
For the native exception case, the conversion evaluates since it is originally a native exception, it will implicitly convert to a java exception with the managed part stripped.

Whereas, since this is a managed exception, it will preserve that part of the trace.